### DISCUSSION TITLE
Advice on subr--template-*.eln files still ending up in eln-cache

### DISCUSSION THREAD

**Discussion by ashlineldridge** [emacscollective/no-littering#206] [Support] [answered]
Hi,

The majority of eln-cache files get moved into `${user-emacs-directory}/var/eln-cache` after following the [advice in the README](https://github.com/emacscollective/no-littering#native-compilation-cache), however I still get a bunch of `subr--trampoline-*.eln` files that get created under `${user-emacs-directory}/eln-cache`. All other `.eln` files get created in the correct location. Any ideas on what the issue might be? Below are the contents of my `${user-emacs-directory}/eln-cache` directory which also show the version of Emacs.

Thanks for any assistance.

```
❯ find eln-cache
eln-cache
eln-cache/29.0.90-e17872ef
eln-cache/29.0.90-e17872ef/subr--trampoline-64657363726962652d6275666665722d62696e64696e6773_describe_buffer_bindings_0.eln
eln-cache/29.0.90-e17872ef/subr--trampoline-6d616b652d70726f63657373_make_process_0.eln
eln-cache/29.0.90-e17872ef/subr--trampoline-68616e646c652d7377697463682d6672616d65_handle_switch_frame_0.eln
eln-cache/29.0.90-e17872ef/subr--trampoline-72656e616d652d66696c65_rename_file_0.eln
```

**Comment by tarsius** [comment]
> Any ideas on what the issue might be?

This is probably due to compilation that begins before that code is evaluated and/or compilation triggered by `emacs -Q`. Personally I have given up on fixing that, but if you figure it out, please let me know.

**Comment by ashlineldridge** [comment]
Will do. Thanks, @tarsius.

It still happens even though the call to `startup-redirect-eln-cache` is the first call in `early-init.el`. And the misplaced `eln-cache` directory only shows at the moment I open my `init.el` file - I've tried opening a bunch of other files including shell files, `early-init.el`, `.gitignore` files, etc, but it only seems to appear when I open `init.el`.

**Comment by ashlineldridge** [comment, marked as answer]
After writing that, I've now narrowed it down to Flycheck! I have the following line in my `init.el`:

```
(setq flycheck-emacs-lisp-load-path 'inherit)
```

When I omit this line (or disable Flycheck altogether), the misplaced `eln-cache` is never created.

**Comment by mornbell** [comment]
```
(when (fboundp 'native-comp-eln-load-path) (setq native-comp-eln-load-path eln-cache-directory))
```
this line could help.
